app/src/main/java/com/github/giommok/softwaredevproject/FirebaseAccount.java

```diff
@@ -5,48 +5,70 @@
 
 import androidx.annotation.NonNull;
 
+import com.github.ravifrancesco.softwaredevproject.POIPoint;
 import com.google.firebase.auth.FirebaseAuth;
 import com.google.firebase.database.DataSnapshot;
 import com.google.firebase.database.DatabaseError;
 import com.google.firebase.database.DatabaseReference;
+import com.google.firebase.database.GenericTypeIndicator;
 import com.google.firebase.database.ValueEventListener;
 
+import org.osmdroid.bonuspack.location.POI;
+
 import java.util.ArrayList;
-import java.util.Arrays;
+import java.util.Collection;
 import java.util.Collections;
 import java.util.HashMap;
 import java.util.HashSet;
+import java.util.List;
 import java.util.Map;
-import java.util.concurrent.Future;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
 
 /* Singleton class containing the only possible account connected */
 public class FirebaseAccount implements Account {
 
-    private static DatabaseReference dbRefUsername = Database.refRoot.child("users/null/username");
-    private static ValueEventListener usernameListener = new ValueEventListener() {
+    private static String username = "null";
+    private static long score = 0;
+    private static FirebaseAccount account = null;
+    private static DatabaseReference dbRefUser = Database.refRoot.child("users/null/username");
```

Comment:
As the reference is to the user, I think it is better to use just the user as child.
```suggestion
    private static DatabaseReference dbRefUser = Database.refRoot.child("users/null");
```